lib/generate-index.js

```diff
@@ -28,52 +28,60 @@ function generateIndex (playbook, pages) {
   if (!pages.length) return {}
   // Map of Lunr ref to document
   const documentsStore = {}
-  const documents = pages.map((page) => {
-    const html = page.contents.toString()
-    const $ = cheerio.load(html)
-
-    // Fetch just the article content, so we don't index the TOC and other on-page text
-    // Remove any found headings, to improve search results
-    const article = $('article.doc')
-    const $h1 = $('h1', article)
-    const documentTitle = $h1.first().text()
-    $h1.remove()
-    const titles = []
-    $('h2,h3,h4,h5,h6', article).each(function () {
-      let $title = $(this)
-      // If the title does not have an Id then Lunr will throw a TypeError
-      // cannot read property 'text' of undefined.
-      if ($title.attr('id')) {
-        titles.push({
-          text: $title.text(),
-          id: $title.attr('id')
-        })
-      }
-      $title.remove()
+  const documents = pages
+    .map((page) => {
+      const html = page.contents.toString()
+      const $ = cheerio.load(html)
+      return { page, $ }
+    })
+    // Exclude pages marked as "noindex"
+    .filter(({ page, $ }) => {
+      const $metaRobots = $('meta[name=robots]')
+      return !(($metaRobots && $metaRobots.attr('content') === 'noindex') || (page.asciidoc && page.asciidoc.noindex === ''))
```

Comment:
@mojavelinux is it the correct way to check if an attribute is defined?

```adoc
= Secret page
:noindex:

This page won't be indexed.
```
```rb
page.asciidoc && page.asciidoc.noindex === '' # trueish
```

Reply:
@mojavelinux Could you please confirm that this is the right way to check if `noindex` attribute is defined?

Apart from that I think this pull request is ready to be merged.

Reply:
It should be `page.asciidoc.attributes.noindex`. Otherwise, yes, you have the right idea.